geode-core/src/main/java/org/apache/geode/distributed/internal/FunctionExecutionPooledExecutor.java

```diff
@@ -222,13 +223,16 @@ protected void beforeExecute(Thread t, Runnable r) {
     if (this.stats != null) {
       this.stats.startJob();
     }
+    ThreadMonitoringUtils.getThreadMonitorObj()
```

Comment:
When I mentioned not using a singleton pattern, I meant that some class such as ClusterDistributionManager would be the original class to create and own the ThreadMonitor. All other classes that use the ThreadMonitor should accept it via its constructor rather than reaching out to get it. This new pattern is very similar to a singleton which and it's difficult to unit test and people resort to using PowerMock to override static methods.

I try to have all dependencies be a final field which is set by passing the instance in via the constructor. The only exception is Logger which is typically a static final field on the class.

Reply:
but i do create and save the ThreadMonitor object in ClusterDistributionManager and this is what u required. i simply created this function to have a single point of reaching the obejct that is stored under ClusterDistributionManager , instead of implementing such method in multiple places which uses it.

Reply:
can u please reply?
this function - ThreadMonitoringUtils.getThreadMonitorObj() simply impelemts the following - 

  public static ThreadMonitoring getThreadMonitorObj() {
    InternalDistributedSystem ds = InternalDistributedSystem.getAnyInstance();
    if (ds == null)
      return threadMonitorDummy;
    DistributionManager distributionManager = ds.getDistributionManager();
    return distributionManager.getThreadMonitoring();
  }

by this i wanted to avoid implementing it in multiple places - but rather have a single access point

Reply:
I think the key point from Kirk's suggestion is that if a class needs to use a `ThreadMonitor`, the reference to `ThreadMonitor` should be passed in on the constructor for that class.  By using DependencyInjection (instead of Singletons) we can make it easier to write testable code.

```
public MyClassThatUsesThreadMonitor {
  private final ThreadMonitor monitor;
  public MyClassThatUsesThreadMonitor(ThreadMonitor monitor) {
    this.monitor = monitor;
  }

  public void doSomething() {
    monitor.XXX();  // avoid singleton call
  }

  ...
}
```

Reply:
requested changes were done